lib/index.js

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const _logger = require('./logger');
+const assert = require('assert-plus');
+const bedrock = require('bedrock');
+const {asyncHandler} = require('bedrock-express');
+const {ensureAuthenticated} = require('bedrock-passport');
+const {create} = require('apisauce');
+const https = require('https');
+const {util: {BedrockError}} = bedrock;
+
+const proxies = [];
+
+// HTTP methods that have a body
+const bodyMethods = new Set(['post', 'put', 'patch']);
+
+exports.addProxy = ({
+  authorizationCheck, method, route, strictSSL = true, targetEndpoint
+}) => {
+  assert.func(authorizationCheck, 'options.authorizationCheck');
+  assert.string(method, 'options.method');
+  assert.string(route, 'options.route');
+  assert.bool(strictSSL, 'options.strictSSL');
+  assert.string(targetEndpoint, 'options.targetEndpoint');
+
+  const baseURL = targetEndpoint;
+  const httpsAgent = new https.Agent({rejectUnauthorized: strictSSL});
+  const targetApi = create({baseURL});
+
+  proxies.push({
+    authorizationCheck,
+    httpsAgent,
+    method,
+    route,
+    targetApi
+  });
+};
+
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  // TODO: conditionally add rateLimiting API?
+  for(const {
+    authorizationCheck, httpsAgent, method, route, targetApi
+  } of proxies) {
```

Comment:
Is this taking destructuring too far?  Seems better to make the loop part easier to read:
```js
for(const proxy of proxies) {
   const {authorizationCheck, httpsAgent, method, route, targetApi} = proxy;
   ...
}
```

Reply:
done.